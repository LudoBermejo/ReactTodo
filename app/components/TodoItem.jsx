import React from 'react';
import moment from 'moment';

const TodoItem = (props) => {
  const { id, value, completed, createdAt, completedAt } = props;

  function renderCreatedAt(createdAt, completedAt) {

    if(!completedAt) {
      return `Created ${moment.unix(createdAt).format('DD MM YYYY | HH:mm')}`;
    } else {
      return `Completed ${moment.unix(completedAt).format('DD MM YYYY | HH:mm')}`;
    }
  }

  function onChangeComplete(e) {
    e.preventDefault();
    props.onChangeComplete(id);
  }
  return (
    <li
      id={`list_item_${id}`}
      onClick={onChangeComplete}
      className="todo-item"
    >
      <input
        type="checkbox"
        checked={completed}
      />
      <p>{value}</p>
      <p>{renderCreatedAt(createdAt, completedAt)}</p>
    </li>
  );
};

TodoItem.propTypes = {
  id: React.PropTypes.string.isRequired,
  value: React.PropTypes.string.isRequired,
  completed: React.PropTypes.bool.isRequired,
  createdAt: React.PropTypes.number.isRequired,
  completedAt: React.PropTypes.number,
  onChangeComplete: React.PropTypes.func.isRequired
};

export default TodoItem;
