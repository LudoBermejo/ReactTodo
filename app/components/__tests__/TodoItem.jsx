import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import TodoItem from 'TodoItem';

describe('TodoItem', function() {

  it('renders properly', function() {
    const staticValue = {
      id: 1,
      value: 'one'
    }

    const todoItem = renderer.create(<TodoItem {...staticValue} />).toJSON();
    expect(todoItem).toMatchSnapshot();
  });

})

