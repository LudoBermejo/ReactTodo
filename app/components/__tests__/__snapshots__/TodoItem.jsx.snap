exports[`TodoItem renders properly 1`] = `
<li
  className="todo-item"
  id="list_item_f6cd808b-ae79-4c7e-8679-6950a7d259e0"
  onClick={[Function]}>
  <input
    checked={false}
    type="checkbox" />
  one
</li>
`;
